windows-gaming-driver/src/config.rs

```diff
@@ -184,6 +190,32 @@ impl From<DeviceId> for (u16, u16) {
     }
 }
 
+impl From<(u16, u16)> for DevicePort {
+    fn from(old: (u16, u16)) -> DevicePort {
+        DevicePort {
+            bus: old.0,
+            port: old.1,
+        }
+    }
+}
+
+impl From<DevicePort> for (u16, u16) {
+    fn from(old: DevicePort) -> (u16, u16) {
+        (old.bus, old.port)
+    }
+}
+
+impl From<DeviceId> for UsbBinding {
+    fn from(old: DeviceId) -> Self {
+        UsbBinding::ById(old)
+    }
+}
+impl From<DevicePort> for UsbBinding {
```

Comment:
Not sure if from impls are ideal here given that wrapping in the enum is not exactly a clear-cut conversion and easy to do anyways.